### PR TITLE
UI/automation/test

### PR DESCRIPTION
**Pull request by kacper-qa**
UI automatic test created for registration and login.

### CODE REVIEW

File: product-backend/scripts/create_db_local_exe.sql
Comment (leszekgu) [inline product-backend/scripts/create_db_local_exe.sql:1]: will catch up with you about this in the morning. Long story short do `git rm example_data_local_exe.sql && git rm create_db_local_exe_sql`

File: ui-test/test/pageobjects/login.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import Page from './page.js';
+
+class LoginPage extends Page {  
+
+async fillLoginForm(email, password) {
```
Comment (leszekgu) [inline ui-test/test/pageobjects/login.ts]: missing indentation

File: ui-test/test/pageobjects/login.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    await $('//*[@id="email"]').setValue(email);
+    await $('//*[@id="password"]').setValue(password);
+
```
Comment (leszekgu) [inline ui-test/test/pageobjects/login.ts]: empty line not needed

File: ui-test/test/pageobjects/register.page.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import Page from './page.js';
+
+export class RegisterPage extends Page {  
+
+
+
+async fillRegistrationForm(email, password, isAdmin: boolean) {
```
Comment (leszekgu) [inline ui-test/test/pageobjects/register.page.ts]: missing indentation

File: ui-test/test/pageobjects/view-job-roles.page.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  async isTitleCorrect() {
+    const title = await browser.getTitle();
+    return title === 'Job Roles'; // Update the expected title
+  }
+
+  async engineeringPage(){
+    await $('//*[@id="orders"]/table/tbody/tr[1]/td/a').click();
+
+  }
+
```
Comment (leszekgu) [inline ui-test/test/pageobjects/view-job-roles.page.ts]: please remove those empty lines, zero or one would be enough

File: ui-test/test/specs/add-and-view-job-role.e2e.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+
+import ViewJobRolePage from '../pageobjects/view-job-roles.page.js'; // Import ViewJobRolePage class
+
+  describe('Job Role Viewing Process', () => {
```
Comment (leszekgu) [inline ui-test/test/specs/add-and-view-job-role.e2e.ts]: here indentation to the left (too many spaces or tabs) and whole describe block (lines 4-10)

File: ui-test/test/specs/register.e2e.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import RegisterPage from '../pageobjects/register.page.js'; 
+import LoginPage from '../pageobjects/login.js'; 
+
+describe('Registration and Login Process', () => {
+    it('Register as Admin || HappyPath', async () => {
+     
+      await RegisterPage.goToRegisterPage();
+      await RegisterPage.fillRegistrationForm('useradmin@kainos.com', 'Test123@', true);
+      await RegisterPage.submitRegistrationForm();
```
Comment (leszekgu) [inline ui-test/test/specs/register.e2e.ts]: where is the 'expect' part?

File: ui-test/test/specs/login.e2e.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import LoginPage from '../pageobjects/login.js'; 
+
+describe('Login & Registration Process', () => {
+  
+    it('Login with valid email and password || HappyPath', async () => {
+     
+      await LoginPage.goToLoginPage();
+      await LoginPage.fillLoginForm('useradmin@kainos.com', 'Test123@');
+      await LoginPage.submitLogin(); 
+      await browser.pause(5000);
```
Comment (leszekgu) [inline ui-test/test/specs/login.e2e.ts]: missing expect part

File: ui-test/test/specs/view-job-roles.e2e.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import viewJobRolesPage from "../pageobjects/view-job-roles.page.js";
+import {browser, expect} from '@wdio/globals';
+
+describe('Get a view of job-roles page', async () => {
+    it('View-job-roles and return to main page', async () => {
+
+        await browser.url("https://wsbz8mje8f.eu-west-1.awsapprunner.com/job-roles");
+        await viewJobRolesPage.isTitleCorrect();
+        await viewJobRolesPage.engineeringPage();
+
```
Comment (leszekgu) [inline ui-test/test/specs/view-job-roles.e2e.ts]: missing expect part

**Review by leszekgu — CHANGES_REQUESTED** [review]
few syntax issues and in many places you've missed 'expect' method

File: product-backend/scripts/create_db_local_exe.sql
Comment (kacper-qa) [reply product-backend/scripts/create_db_local_exe.sql:1]: Hi, 

From specs below files have been deleted:
- view-job-roles, 
- login, 
- register
- add-and-view-job-roles
From page objects below file has been deleted:
- view-job-roles

Instead we got one specs file "register_login_user" for test cases HappyPath, unHappyPath for registration and login e2e.

**Review by leszekgu — APPROVED** [review]

**Pull request merged**